### PR TITLE
FileObject

### PR DESCRIPTION
**Pull request by benrudolph**
@snopoke @czue this has been on my backlog for a while. This is the beginnings of trying to fix our redis queries that take 2+ minutes. I want to make a `FileObject` class that talks like the `CachedObject` so we can interchange them. That way we can use the `FileObject` to store a file like `media_profile` in the temp_dir on the NFS instead of in Redis. This PR is a pretty long way off in terms of cleanliness, I want to rejig some of the inheritance so it makes more sense to have the FileSystem as a backend instead of Redis. Let me know if you think this is worthwhile to pursue

buddy: @orangejenny

### CODE REVIEW

**Comment by snopoke** [comment]
I think this is a good move. We should also make sure that these requests get a `TransferHttpResponse` so that nginx can handle the file streaming.

**Comment by benrudolph** [comment]
agreed

File: dimagi/utils/django/cached_object.py [hunk 1]
```diff
@@ -271,6 +296,83 @@ def get_all_keys(self):
...
+            f.write(simplejson.dumps(object_meta.to_json()))
+
+    def get_all_keys(self):
+        """
+        Returns all FULL keys
+        """
+        if not self.base_dir:
+            return [], []
+        full_stream_keys = glob.glob(os.path.join(self.base_dir, self.stream_key(WILDCARD)))
+        full_meta_keys = glob.glob(os.path.join(self.base_dir, self.meta_key(WILDCARD)))
```
Comment (snopoke) [inline dimagi/utils/django/cached_object.py]: Is it necessary to get all the keys here? Can't you just check for the existence of one and then assume it cached. If a file is missing later you could raise an exception that triggers a re-cache.

Comment (snopoke) [inline dimagi/utils/django/cached_object.py]: Just thinking that this might be a bit slow

**Comment by czue** [comment]
not reviewing too closely but generally concept sounds good to me

File: dimagi/utils/django/cached_object.py (hunk 1 above)
Comment (benrudolph) [inline dimagi/utils/django/cached_object.py]: yea i agree. i actually think there will only ever be one cache key. the sizes are used for `CachedImage`. so i think we can minimally just call `os.path.exists(stream_key)` and that should tell us the correct information

**Pull request closed without merging**